File: src/main/java/com/nndmove/app/domain/History.java
package com.nndmove.app.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.io.Serializable;
import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

/**
 * A History.
 */
@Entity
@Table(name = "history")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
@SuppressWarnings("common-java:DuplicatedBlocks")
public class History implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
    @SequenceGenerator(name = "sequenceGenerator")
    @Column(name = "id")
    private Long id;

    @NotNull
    @Column(name = "part", nullable = false)
    private Integer part;

    @NotNull
    @Column(name = "stop_time", nullable = false)
    private Integer stopTime;

    @ManyToOne(fetch = FetchType.LAZY)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties(value = { "genres" }, allowSetters = true)
    private Movie movie;

    // jhipster-needle-entity-add-field - JHipster will add fields here

    public Long getId() {
        return this.id;
    }

    public History id(Long id) {
        this.setId(id);
        return this;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getPart() {
        return this.part;
    }

    public History part(Integer part) {
        this.setPart(part);
        return this;
    }

    public void setPart(Integer part) {
        this.part = part;
    }

    public Integer getStopTime() {
        return this.stopTime;
    }

    public History stopTime(Integer stopTime) {
        this.setStopTime(stopTime);
        return this;
    }

    public void setStopTime(Integer stopTime) {
        this.stopTime = stopTime;
    }

    public User getUser() {
        return this.user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public History user(User user) {
        this.setUser(user);
        return this;
    }

    public Movie getMovie() {
        return this.movie;
    }

    public void setMovie(Movie movie) {
        this.movie = movie;
    }

    public History movie(Movie movie) {
        this.setMovie(movie);
        return this;
    }

    // jhipster-needle-entity-add-getters-setters - JHipster will add getters and setters here

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof History)) {
            return false;
        }
        return getId() != null && getId().equals(((History) o).getId());
    }

    @Override
    public int hashCode() {
        // see https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/
        return getClass().hashCode();
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "History{" +
            "id=" + getId() +
            ", part=" + getPart() +
            ", stopTime=" + getStopTime() +
            "}";
    }
}

File: src/main/java/com/nndmove/app/domain/History.kt
package com.nndmove.app.domain

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import java.io.Serial
import java.io.Serializable

/**
 * A History.
 */
@Entity
@Table(name = "history")
class History : Serializable {
    // jhipster-needle-entity-add-field - JHipster will add fields here
    @JvmField
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
    @SequenceGenerator(name = "sequenceGenerator")
    @Column(name = "id")
    var id: Long? = null

    @JvmField
    @Column(name = "part", nullable = false)
    var part: @NotNull Int? = null

    @JvmField
    @Column(name = "stop_time", nullable = false)
    var stopTime: @NotNull Int? = null

    @JvmField
    @ManyToOne(fetch = FetchType.LAZY)
    var user: User? = null

    @JvmField
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties(value = ["genres"], allowSetters = true)
    var movie: Movie? = null

    fun id(id: Long?): History {
        this.id = id
        return this
    }

    fun part(part: Int?): History {
        this.part = part
        return this
    }

    fun stopTime(stopTime: Int?): History {
        this.stopTime = stopTime
        return this
    }

    fun user(user: User?): History {
        this.user = user
        return this
    }

    fun movie(movie: Movie?): History {
        this.movie = movie
        return this
    }

    // jhipster-needle-entity-add-getters-setters - JHipster will add getters and setters here
    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is History) {
            return false
        }
        return (id != null) && this.id == o.id
    }

    override fun hashCode(): Int {
        // see https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/
        return javaClass.hashCode()
    }

    // prettier-ignore
    override fun toString(): String {
        return "History{" +
                "id=" + id +
                ", part=" + part +
                ", stopTime=" + stopTime +
                "}"
    }

    companion object {
        @Serial
        private val serialVersionUID = 1L
    }
}
